var/spack/repos/builtin/packages/neurodamus/package.py

```diff
@@ -83,41 +94,78 @@ def syn2_dep_libs(self):
+
+        include_flag += ' -I%s -I%s %s' % (spec['reportinglib'].prefix.include,
+                                           spec['hdf5'].prefix.include,
+                                           profile_flag)
+        link_flag += ' %s -L%s -lhdf5 -L%s -lz' % (
+                     spec['reportinglib'].libs.ld_flags,
+                     spec['hdf5'].prefix.lib,
+                     spec['zlib'].prefix.lib)
+
+        nrnivmodl = which('nrnivmodl')
+        with profiling_wrapper_on():
             nrnivmodl('-incflags', include_flag,
-                      '-loadflags', link_flag,
-                      modlib)
+                      '-loadflags', link_flag, 'm')
+        # special exists or fail
+        special = os.path.join(os.path.basename(self.neuron_archdir), 'special')
+        assert os.path.isfile(special)
+
+    def install(self, spec, prefix):
+        """ Move libs to destination.
+            Libs are sym-linked. Compiled libs into libs, special into bin
+        """
+        neurodamus_base = spec['neurodamus-base'].prefix
+        arch = os.path.basename(self.neuron_archdir)
+        os.makedirs(prefix.lib.modc)
+        os.makedirs(prefix.bin)
+
+        os.symlink(neurodamus_base.lib.hoclib, prefix.lib.hoclib)
+        os.symlink(neurodamus_base.lib.modlib, prefix.lib.modlib)
+
+        if os.path.isdir(neurodamus_base.python):
+            os.symlink(neurodamus_base.python, prefix.python)
+
+        shutil.move(os.path.join(arch, 'special'), prefix.bin)
+
+        # Copy c mods
+        for cmod in find(arch, "*.c", recursive=False):
+            shutil.move(cmod, prefix.lib.modc)
 
-    @run_after('install')
-    def check_install(self):
-        special = '%s/special' % join_path(self.prefix, os.path.basename(self.neuron_archdir))
-        if not os.path.isfile(special):
-            raise RuntimeError("Neurodamus installion check failed, %s not created!" % special)
+        # Handle non-binary special
+        if os.path.exists(arch + "/.libs/libnrnmech.so"):
+            shutil.move(arch + "/.libs/libnrnmech.so", prefix.lib)
+            sed = which('sed')
+            sed('-i', 's#-dll .*#-dll %s#' % prefix.lib.join('libnrnmech.so'), prefix.bin.special)
 
     def setup_environment(self, spack_env, run_env):
-        run_env.prepend_path('PATH', join_path(self.prefix, os.path.basename(self.neuron_archdir)))
-        run_env.set('HOC_LIBRARY_PATH', self.spec['neurodamus-base'].prefix.lib.hoclib)
+        run_env.prepend_path('PATH', self.prefix.bin)
+        run_env.set('HOC_LIBRARY_PATH', self.prefix.lib.hoclib)
+        if os.path.isdir(self.prefix.python):
+            for m in spack_env.env_modifications:
+                if m.name == 'PYTHONPATH':
+                    run_env.prepend_path('PYTHONPATH', m.value)
+            run_env.prepend_path('PYTHONPATH', self.prefix.python)
+            run_env.set('PYDAMUS', self.prefix.python.join('init.py'))
```

Comment:
Currently people have : 

```
mpirun ... $HOC_LIBRARY_PATH/init.hoc
```

I like following for consistency:

```
mpirun ... $PYTHON_LIBRARY_PATH/init.hoc
```

Instead of `mpirun ... $PYDAMUS`. So may be we can set something like PYTHON_LIBRARY_PATH or similar instead of PYDAMUS?